sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLogSuite.scala

```diff
@@ -209,14 +209,20 @@ class HDFSMetadataLogSuite extends SparkFunSuite with SharedSQLContext {
     }
 
     // Open and delete
-    val f1 = fm.open(path)
-    fm.delete(path)
-    assert(!fm.exists(path))
-    intercept[IOException] {
-      fm.open(path)
+    if (Utils.isWindows) {
+      fm.open(path).close()
+      fm.delete(path)
+      fm.delete(path) // should not throw exception
```

Comment:
Windows holds an exclusive lock so it does not allow to remove a file when it is open. So, I re-wrote the test cases for WIndows.

cc @tdas, do you mind if I ask to see whether these tests on Windows are fine?